### DISCUSSION TITLE
Add ability to use AL2 as base OS for Stackstorm (ST2)

### DISCUSSION THREAD

**Discussion by dumbworker** [StackStorm/st2#6016] [Ideas]
Project Overview:
The goal of the project is to enable StackStorm to run on AL2 OS from its existing Ubuntu base OS. The enablement will involve updating the dockerfile, Python version, and dependent Python libraries to ensure seamless compatibility and performance on AL2. Set up CI/CD for ST2 on AL2, so that AL2-based ST2 is available over ST2 distribution channels. The effort will also include comprehensive testing and validation to verify the stability and reliability of the platform on the AL2.
Benefits of Porting to Amazon Linux 2 (AL2) OS:

1. Enhanced Performance and Optimization: AL2 is designed to provide better performance, higher efficiency, and optimization for running applications on AWS infrastructure. This enablement will improve overall performance for StackStorm deployments in AWS environments.
2. Deployment in Hardened Environments: AL2 meets multiple regulatory requirements for secure and air-gapped deployments. AL2 is approved and used in environments certified at various Impact Levels. The Amazon products are listed here - https://marketplace.fedramp.gov/products/F1603047866
3. Security and Maintenance: Amazon Linux 2 offers long-term support with regular security updates, ensuring that the base OS used for StackStorm remains secure and protected against potential vulnerabilities.
4. AWS Integration: It will enable better integration with AWS services, making it easier to leverage various AWS features within the StackStorm workflows.
5. Cost-Effectiveness: AL2 is available at no additional cost.

Project Phases:
The project can be divided into the following key phases:

1. Development: Modify StackStorm components to ensure compatibility with AL2 and implement any necessary changes.
2. Setup build and release pipelines: Add CI/CD for Stackstorm on AL2 for release in line with existing build infrastructure.
3. Testing and Validation: Conduct comprehensive testing to verify the functionality and performance of StackStorm on AL2.
4. Documentation and Community Support: Create comprehensive documentation to guide users in migrating and deploying StackStorm on AL2. Provide support and assistance to the community during and after the migration.

Enabling StackStorm to run on Amazon Linux 2 presents an exciting opportunity to enhance the platform’s performance, security, and integration capabilities on AWS. The project aligns StackStorm’s commitment to providing a robust and efficient open-source event-driven automation solution for the community. We seek your approval and support for this project, as it will significantly contribute to the growth and success of StackStorm and its adoption in the AWS ecosystem. We are open to discussing further details and project planning.
Thank you for considering this proposal, and we eagerly await your valuable input.

**Comment by arm4b** [comment]
Conceptually, I think it's a great idea!

Technically, Python version matrix was the hardest blocker for adding different Linux distros, see https://github.com/StackStorm/community/issues/103. Amazon Linux 2 supports python 3.8 (which is supported by StackStorm), and so it should be doable, but worth doing more research about the technical corner cases.

The hard part is the development and then the hardest is maintenance. At this moment the project is having a hard time maintaining even the core basics.
Also, if there is a little popularity behind the flavor, it would be a difficulty to find volunteers to fix something related to the builds.
@dumbworker This might be a long project indeed. Do you have engineering cycles helping adding this to the ST2?
Curious if there are more folks interested in taking part contributing this. @khushboobhatia01 ?

**Reply by dumbworker** [reply]
Hello @armab 
Yes, support and maintenance would be hard. We do have this running in a state better than POC. So we are confident of completing this.
Your support and guidance would be of great help to plan support and maintenance. We could meet to discuss this and understand a possible path around support and maintenance.

**Reply by arm4b** [reply]
Ah, so it means you have lots of things already working for Amazon Linux 2 on your side and would like to contribute the work you already did to the upstream.
That's definitely the right way to do it 👍 

@dumbworker  Could you folks join us in the next monthly StackStorm Community TSC meeting?
Based on https://github.com/StackStorm/community/issues/33 calendar, it should be on September, 12th.

**Reply by dumbworker** [reply]
Hello @armab 
Sorry I could not join on the 12, We will join the next StackStorm Community TSC meeting

**Reply by arm4b** [reply]
@dumbworker Awesome! It will be also nice if you could bring more attention to this Issue in the `#tsc` channel of Slack, - it's helpful to have more 👀 and reminders for people to take a look.

**Comment by nzlosh** [comment]
What is meant by:

> to enable StackStorm to run on AL2 OS from its existing Ubuntu base OS

Is the intention to replace the Ubuntu base in github actions with AL2 or to add AL2 to the test matrix as an additionally supported distribution in CircleCI package testing?

**Reply by dumbworker** [reply]
Hello @nzlosh 
We do not want to replace Ubuntu base. We would like to add AL2 as additional distribution.

**Comment by winem** [comment]
Why AL2 and not AL2023 which is the (more or less) Fedora based successor of AL2? I think it makes sense to focus on AL2023 if we want to support an Amazon Linux distribution but I might miss something.

**Reply by setswei** [reply]
I agree that reviewing AL2023 should also be considered as AL2 was slated to be EOL in June this year. But due to massive pushback from the Amazon user base they have extended that EOL date by 2 more years.
